src/modules/usrloc/udomain.c

```diff
@@ -909,12 +910,22 @@ int db_timer_udomain(udomain_t* _d)
 	vals[1].nul = 0;
 	UL_DB_EXPIRES_SET(&vals[1], 0);
 
+	keys[2] = &srv_id_col;
+	ops[2] = OP_EQ;
+	vals[2].type = DB1_INT;
+	vals[2].nul = 0;
+	vals[2].val.int_val = server_id;
+
 	if (ul_dbf.use_table(ul_dbh, _d->name) < 0) {
 		LM_ERR("use_table failed\n");
 		return -1;
 	}
+	
+	if (ul_db_srvid != 0) {
+		key_num = 3;
+	}
 
```

Comment:
I think it is better to move this `if` bloc at line 913 above and have those assignements for keys/ops/vals with index `[2]` inside it. It is better to see have related operations grouped together, avoiding also useless operations if ul_db_srvid is 0.

Reply:
Agree. Done.